design-documents/graph-ql/coverage/customer-email-password-update.md

```diff
@@ -0,0 +1,71 @@
+## Separate mutations for updating customer email and password
+
+### Problem overview
+`createCustomer` and `updateCustomer` operations are currently sharing `CustomerInput` object. Among other fields `CustomerInput` contains `email` and `password`.
+
+`password` has different semantics in these two mutations:
+ - In scope of `createCustomer` it is a password of the newly created user
+ - In scope of `updateCustomer` it serves for verification of the user's current password, when changing `email`
+
+### Proposed solution
+
+**updateCustomerV2**
+
+Deprecate `updateCustomer` mutation in favor of `updateCustomerV2`. `CustomerUpdateInput` is similar to `CustomerInput` but excludes `password` and `email` fields.
+
+ ```graphql
+ mutation {
+    updateCustomerV2(input: CustomerUpdateInput!): CustomerUpdateOutput
+ }
+ 
+ type CustomerUpdateOutput {
+    customer: Customer!
+ }
+ 
+ type CustomerUpdateInput {
+     date_of_birth: String
+     dob: String
+     firstname: String
+     gender: Int
+     is_subscribed: Boolean
+     lastname: String
+     middlename: String
+     prefix: String
+     suffix: String
+     taxvat: String
+ }
+ ```
+ 
+ **updateCustomerEmail**
+ 
+ ```graphql
+  mutation {
+     updateCustomerEmail(email: String!, current_password: String!): UpdateCustomerEmailOutput
+  }
+  
+   type UpdateCustomerEmailOutput {
+      customer: Customer!
+   }
+ ```
+ 
+ **updateCustomerPassword**
+ 
+ ```graphql
+  mutation {
+     updateCustomerPassword(password: String!, current_password: String!): UpdateCustomerPasswordOutput
```

Comment:
I would suggest to rename current_password to old_password, to make it less confusing. current_password might be interpreted as new password.

Reply:
Adjusted, thank you.